### ISSUE TITLE
Create a check for if an object is a JSON or NWIS Response object in extract_nwis_df

### ISSUE THREAD

**Issue by nkorinek** [mroberge/hydrofunctions#37]
Not an issue, just a recommendation! Recently, hydrofunctions extract_nwis_df changed to accept a JSON of a response instead of the raw response object. The line to turn the response to a JSON is still in the code, but commented out. It would possibly easier to use the function if it took either a JSON or a raw response. A simple type check at the top should make that doable. Something like: 

```
if type(nwis_dict) is not dict:
    nwis_dict = nwis_dict.json()
```

Just a thought!

**Comment by mroberge** [comment]
This sounds good to me.
Do you want me to make the change, or do you want to try your hand at a pull request?

- fork hydrofunctions (or update the fork you already have)
- create a new branch off of Develop
- make your edits & reference this issue in your commit message
- Add some tests!
    - try running the tests I already set up (type `python setup.py test` from the same directory as the setup.py file)
    - copy a test that already exists, but feed it a dict and check that it still extracts a dataframe
    - create another test and feed it a fake response object. check that it still extracts a dataframe
- submit your pull request!

**Comment by nkorinek** [comment]
I'll try it out and see how it goes!

**Comment by mroberge** [comment]
Great! @nkorinek 

One thing that most people don't seem to know about is that you can submit  your pull request almost immediately after creating your new branch. You don't have to wait until you've actually added any new code. Most people like to wait until everything is perfect, but the problem with that is that nobody knows that you are working on something, so you can get duplication. Think of two outfielders in baseball going for the same fly ball. They start talking to each other BEFORE they catch the ball, so they don't both try to catch it, or both assume the other person has it... (sportsball analogies aren't my personal strength)

Also, if you submit the PR right away, I can give you suggestions if you get stuck on something.

Good luck!
-Marty

**Comment by nkorinek** [comment]
Sounds good! I'll open that up as soon as possible, probably over the weekend sometime.

**Comment by mroberge** [comment]
Closed with (cf10f95)[https://github.com/mroberge/hydrofunctions/commit/cf10f95446e4b9254438e9373be4c7c08120b094].

**Comment by nkorinek** [comment]
Thanks @mroberge, sorry I didn't get a chance to work on it!
